runtime/vm/jvminit.c

```diff
@@ -2963,65 +3035,127 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
 #else
 			vm->extendedRuntimeFlags &= ~(UDATA)J9_EXTENDED_RUNTIME_REDUCE_CPU_MONITOR_OVERHEAD;
 #endif
-		} else if (0 == strcmp(testString, VMOPT_XXREDUCECPUMONITOROVERHEAD)) {
+		} else if (noReduceCPUMonitorOverhead < reduceCPUMonitorOverhead) {
 			vm->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_REDUCE_CPU_MONITOR_OVERHEAD;
-		} else if (0 == strcmp(testString, VMOPT_XXENABLECPUMONITOR)) {
+		}
+	}
+
+	{
+		IDATA enableCPUMonitor = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXENABLECPUMONITOR, NULL);
+		IDATA disableCPUMonitor = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXDISABLECPUMONITOR, NULL);
+		if (enableCPUMonitor > disableCPUMonitor) {
 			vm->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_ENABLE_CPU_MONITOR;
-		} else if (0 == strcmp(testString, VMOPT_XXDISABLECPUMONITOR)) {
+		} else if (enableCPUMonitor < disableCPUMonitor) {
 			vm->extendedRuntimeFlags &= ~(UDATA)J9_EXTENDED_RUNTIME_ENABLE_CPU_MONITOR;
-		} else if (0 == strcmp(testString, VMOPT_XXRESTRICTCONTENDED)) {
+		}
+	}
+
+	{
+		IDATA restrictContended = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXRESTRICTCONTENDED, NULL);
+		IDATA noRestrictContended = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNORESTRICTCONTENDED, NULL);
+		if (restrictContended > noRestrictContended) {
 			vm->extendedRuntimeFlags &= ~(UDATA)J9_EXTENDED_RUNTIME_ALLOW_APPLICATION_CONTENDED_FIELDS;
-		} else if (0 == strcmp(testString, VMOPT_XXNORESTRICTCONTENDED)) { /* enabling application contended fields implicitly turns on bootstrap contended fields */
+		} else if (restrictContended < noRestrictContended) {
 			vm->extendedRuntimeFlags |= (UDATA)J9_EXTENDED_RUNTIME_ALLOW_APPLICATION_CONTENDED_FIELDS;
 			vm->extendedRuntimeFlags |= (UDATA)J9_EXTENDED_RUNTIME_ALLOW_CONTENDED_FIELDS;
-		} else if (0 == strcmp(testString, VMOPT_XXNOCONTENDEDFIELDS)) { /* disabling bootstrap contended fields implicitly turns off application contended fields  */
+		}
+	}
+
+	if (FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXREDUCECPUMONITOROVERHEAD, NULL) != -1) {
```

Comment:
This is already checked above.

Reply:
Strange.. not too sure why this was here. Sorry about that